src/main/kotlin/calculator/Operation.kt

```diff
@@ -0,0 +1,42 @@
+package calculator
+
+class Operation(
+    private val operandA: Operand
+) {
+    companion object {
+        val Empty = Operation(Scalar(0)).with(AddOperator)
+    }
+
+    private var operator: Operator? = null
+
+    fun with(operator: Operator): Operation {
+        assertOperatorIsNull()
+        this.operator = operator
+        return this
+    }
+
+    fun with(operandB: Operand): Operation {
+        val operator = this.operator.orThrow()
+        return Operation(operator.operate(operandA, operandB))
+    }
+
+    val result: Scalar get() {
+        if (operator != null) {
+            throw IllegalArgumentException("The operation is invalid")
+        }
+        return Scalar(operandA)
+    }
```

Comment:
IntelliJ IDEA의 코드 자동 정렬 기능(⌥⌘L, Ctrl+Alt+L)을 사용하면 더 깔끔한 코드를 볼 수 있을 거예요.

```suggestion
    val result: Scalar
        get() {
            if (operator != null) {
                throw IllegalArgumentException("The operation is invalid")
            }
            return Scalar(operandA)
        }
```